hamilton/function_modifiers/macros.py

```diff
@@ -876,3 +876,134 @@ def optional_config(self) -> Dict[str, Any]:
+        :param _chain: Whether to chain the first parameter. This is the only mode that is supported. Furthermore, this is not externally exposed. @flow will make use of this.
+        """
+        self.transforms = transforms
+        self.collapse = collapse
+        self.chain = _chain
+        self.namespace = namespace
+
+        if self.collapse:
+            raise NotImplementedError(
+                "Collapsing step() functions as one node is not yet implemented for pipe(). Please reach out if you want this feature."
+            )
+
+        if self.chain:
+            raise NotImplementedError("@flow() is not yet supported -- this is ")
+
+    def transform_dag(
+        self, nodes: Collection[node.Node], config: Dict[str, Any], fn: Callable
+    ) -> Collection[node.Node]:
+        """Transforms the subDAG by getting the injectable parameters (anything not
+        produced by nodes inside it), then calling the inject_nodes function on it.
+
+        :param nodes:
+        :param config:
+        :param fn:
+        :return:
+        """
+
+        # We want the output of post-pipe to connect downstream, so the output needs to be named the
+        # same as the original function. We rename the first function to fn_raw and bind it as the
+        # first parameter. When the pipe nodes are wired through we rename the last node as the
+        # original function name
+        original_node = nodes[-1].copy_with(name=f"{nodes[-1].name}_raw")
+        nodes_to_inject = self.inject_nodes({original_node.name: original_node.type}, config, fn)
+        last_node = nodes_to_inject[-1].copy_with(name=f"{nodes[-1].name}")
+
+        # This is a problem since we loose the typehinting of the last node
+        print(last_node.type)
+        out = nodes[:-1]
+        out.append(original_node)
+        out.extend(nodes_to_inject[:-1])
+        out.append(last_node)
+        return out
+
+    def inject_nodes(
+        self, params: Dict[str, Type[Type]], config: Dict[str, Any], fn: Callable
+    ) -> List[node.Node]:
+        """Injects nodes into the graph. This creates a node for each pipe() step,
+        then reassigns the inputs to pass it in."""
+
+        # I think the better approach would be to draw from the last node another `identity` node
+        # that has the correct name but just passes the value of the last pipe node through. This
+        # would allow us to display all pipe transforms in the DAG, otherwise the last one gets
+        # re-named and we effectively hide a node, but I am stuck at how to dynamically add the
+        # correct typehints
+        def __identity(foo: Any) -> Any:
+            return foo
+
+        self.transforms = self.transforms + (step(__identity).named(fn.__name__),)
+
+        # since the dict is always length 1 just gets single value out
+        current_param = next(iter(params))
+        fn_count = Counter()
+        nodes = []
+        for applicable in self.transforms:
```

Comment:
This code looks correct, and I think you'll be able to share with `@pipe` a little later on (it was copies, right?)